src/storage/conversion/ErrorToTemplateConverter.ts

```diff
@@ -25,50 +23,43 @@ import { TypedRepresentationConverter } from './TypedRepresentationConverter';
 export class ErrorToTemplateConverter extends TypedRepresentationConverter {
   private readonly engine: TemplateEngine;
   private readonly templatePath: string;
-  private readonly descriptions: string;
-  private readonly contentType: string;
   private readonly extension: string;
+  private readonly contentType: string;
 
-  public constructor(engine: TemplateEngine, templatePath: string, descriptions: string, contentType: string,
-    extension: string) {
+  public constructor(engine: TemplateEngine, templatePath: string, extension: string, contentType: string) {
     super(INTERNAL_ERROR, contentType);
     this.engine = engine;
-    this.templatePath = resolveAssetPath(templatePath);
-    this.descriptions = resolveAssetPath(descriptions);
-    this.contentType = contentType;
+    this.templatePath = templatePath;
     this.extension = extension;
+    this.contentType = contentType;
   }
 
   public async handle({ representation }: RepresentationConverterArgs): Promise<Representation> {
+    // Obtain the error from the representation stream
     const errors = await arrayifyStream(representation.data);
     if (errors.length !== 1) {
       throw new InternalServerError('Only single errors are supported.');
     }
     const error = errors[0] as Error;
 
-    // Render the template
-    const { name, message, stack } = error;
-    const description = await this.getErrorCodeMessage(error);
-    const variables = { name, message, stack, description };
-    const template = await fsPromises.readFile(this.templatePath, 'utf8');
-    const rendered = this.engine.apply(template, variables);
-
-    return new BasicRepresentation(rendered, representation.metadata, this.contentType);
-  }
-
-  private async getErrorCodeMessage(error: Error): Promise<string | undefined> {
+    // Render the error description using an error-specific template
+    let description: string | undefined;
     if (HttpError.isInstance(error)) {
-      let template: string;
       try {
-        const fileName = `${error.errorCode}${this.extension}`;
-        assert(/^[\w.-]+$/u.test(fileName), 'Invalid error template name');
-        template = await fsPromises.readFile(joinFilePath(this.descriptions, fileName), 'utf8');
+        const templateFile = `${error.errorCode}${this.extension}`;
+        assert(/^[\w.-]+$/u.test(templateFile), 'Invalid error template name');
+        description = await this.engine.render((error.details ?? {}) as NodeJS.Dict<string>,
+          { templateFile, templatePath: this.templatePath });
```

Comment:
The one (minor) issue I have with this solution is that this line is sort of confusing. Specifically: it is calling an engine that has a fixed template, but that fixed template has no influence at all on the result being produced here. In fact, any engine could have been used here, could also have been the one used for pod provisioning. So the fact that which engine it is has no impact here feels a bit weird.

I do completely get the architectural advantages though. This is also going to help for merging the two templating solutions we have now. But it slightly bugs me that we have a function here that is completely independent of the object that contains it if you call it with 2 parameters (functional programming style), but not if you call it with 1 parameter (OOP style).

I don't really have a solution though besides making the 2nd parameter required, but that sort of ruins the main goal of this PR.

Reply:
> Specifically: it is calling an engine that has a fixed template, but that fixed template has no influence at all on the result being produced here.

I understand; I had the same consideration.

Thoughts I had:
- We should really be passing two template engines (can be the same during instantiation)
- The `ErrorToTemplateConverter` should actually be split in two parts, one for the details and one for the templates.
- The real issue is that we don't have parent templates

> In fact, any engine could have been used here, could also have been the one used for pod provisioning. So the fact that which engine it is has no impact here feels a bit weird.

Well, you don't know as a consumer how the template engine works. It might still do things with the default template.

> function here that is completely independent of the object that contains

It's not though; there's still the polymorphism (what kind of engine), and any given engine might have settings that the consumer does not know about.

This last bit was what inspired me to do this PR: for cases where the template is always constant, having to pass the same template every time seems like an implementation detail leaking.

> I don't really have a solution though besides making the 2nd parameter required, but that sort of ruins the main goal of this PR.

I think it still makes sense to have a default template; consumers don't need to know about templates unless they want to get specific.